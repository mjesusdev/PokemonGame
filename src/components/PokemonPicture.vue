<template>
  <div class="pokemon-container mt-4 mb-4">
    <img v-if="!showPokemon" 
      :src="imgSrc"
      class="img-fluid hidden-pokemon"
      alt="pokemon">

    <img v-else
      :src="imgSrc" 
      class="img-fluid fade-in"
      alt="pokemon">
  </div>
</template>

<script>
    // Number, required

    export default {

      props: {
        pokemonId: {
            type: Number,
            required: true
        },
        showPokemon: {
            type: Boolean,
            required: true,
            default: false
        }
      },

      computed: {
        imgSrc () {
          return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${ this.pokemonId }.svg`
        }
      }
    }
</script>

<style scoped>
  /* Pokemon Picture */
  .pokemon-container {
    height: 200px;
    /* width: 400px; */
    margin: 0 auto;
    position: relative;
  }

  img {
    height: 200px;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
  }

  .hidden-pokemon {
      filter: brightness(0);
  }
</style>