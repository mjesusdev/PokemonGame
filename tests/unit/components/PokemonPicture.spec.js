import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture component', () => {
    test('should match with snapshot', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 1,
                showPokemon: false
            }
        })

        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('should show the hide image and the pokemon 100', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })

        const [ img1, img2 ] = wrapper.findAll('img')

        expect( img1.exists() ).toBeTruthy()
        expect( img2 ).toBe(undefined)

        expect( img1.classes('hidden-pokemon') ).toBe(true)

        // https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg

        expect( img1.attributes('src') ).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

    })

    test('should show the pokemon if (showPokemon == true)', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })

        const img1 = wrapper.find('img')

        expect( img1.exists() ).toBeTruthy()

        expect( img1.classes('hidden-pokemon') ).toBe(false)
        expect( img1.classes('fade-in') ).toBe(true)
    })

})