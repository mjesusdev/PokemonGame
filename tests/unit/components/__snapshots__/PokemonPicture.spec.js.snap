// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component should match with snapshot 1`] = `<div class="pokemon-container mt-4 mb-4"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="img-fluid hidden-pokemon" alt="pokemon"></div>`;
